Added before dependents to Modules (#6080)
eclipse/jetty.project: documentation/jetty-documentation/src/main/asciidoc/operations-guide/modules/modules.adoc

The order of the enabled modules is used to determine the processing of the configuration, for example the order of processing of the xref:og-modules-directive-files[+[files]+] section, the order of processing of XML files defined in the xref:og-modules-directive-xml[+[xml]+] section, etc.

[[og-modules-directive-after]]
===== [after]

This directive indicates that this module is ordered after the listed module names, if they are enabled.

For example, module `https` is `[after]` module `http2`.

> Enabling the `https` modules _does not_ enable the `http2` module.

Reviewer: ```suggestion
Enabling the `https` module _does not_ enable the `http2` module.
```

> However, if the `http2` module is enabled (explicitly or transitively), then the `https` module is xref:og-modules-directive-depends[sorted] _after_ the `http2` module.

Reviewer: ```suggestion
However, if the `http2` module is then subsequently enabled (explicitly or transitively), then the `https` module is xref:og-modules-directive-depends[sorted] _after_ the `http2` module.
```

Author: It is not subsequently.  It may already be enabled.

Reviewer: OK, yes.

In this way, you are guaranteed that the `https` module is processed after the `http2` module.

[[og-modules-directive-before]]
===== [before]

This directive indicates that this module is ordered before the listed module names, if they are enabled.

> For example, in order to create a custom `org.eclipse.jetty.server.Server` subclass instance, to which the standard configuration is applied, without modifying existing module files or XML files, you can create a `custom-server` xref:og-modules-custom[Jetty custom module]:

Reviewer: ```suggestion
One use of this ordering could be to ensure that an xml configuration file is executed before any that are associated with the modules listed in the `before` clause. For example: in order to create a custom `org.eclipse.jetty.server.Server` subclass instance without needing to modify existing module or XML files, and to which the standard configuration is applied,  you can create a `custom-server` xref:og-modules-custom[Jetty custom module]:
```
